pullrequest.rst

```diff
@@ -66,6 +66,7 @@ You should have already :ref:`set up your system <setup>`,
 
 * Create a new branch in your local clone::
 
+     git fetch upstream # upstream must be configured as a remote
```

Comment:
Let's move this to a separate bullet above "Create a new branch in your local clone":

* Update the ``upstream`` remote tracking branch:: 

       git fetch upstream

[Reference](https://git-scm.com/docs/git-fetch#_examples)

Reply:
Thanks for the further improvement!  Your suggestion brought up some questions:

The git documentation uses a hyphen for "remote-tracking branch", so maybe we should do that here for consistency?

Was the Reference link for my benefit, or should "remote-tracking branch" be a link in the bullet point?

I found two other places in the guide where the term is just "remote", but it refers to this tracking branch.  They are [section 31.2](https://devguide.python.org/gitbootcamp/#cloning-the-forked-cpython-repository) and [section 1.2.6](https://devguide.python.org/setup/#get-the-source-code).  Should we should expand those to match this bullet point?  If so, would that be another PR?

I appreciate all help and guidance here as I'm still kind of learning the ropes.

Reply:
Thanks @zthompson47. Sure, please put in the hyphen. As an FYI, it updates all of the branches on the remote, but I think in this case keeping things simple is good since some of this can be changed in the configuration of git and folks use different workflows.

Yes the reference was just an FYI for you.

As for the other sections, I typically recommend creating a new PR when working on additional items (i.e. keep a PR small and focused). I believe the usage of remote is correct in these cases since the remote refers to the upstream repository. See https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes and the section on "Fetching and Pulling from Your Remotes".

On further thought, perhaps it would be better to change the text in this PR to something more general like: `Update data from your upstream repository:`. What do you think?

Reply:
Thanks again for the insightful help and reference links.  Now it's more clear that a remote is just a way to reference a remote repository, and a tracking branch is a local branch linked to a branch in a remote.

I agree that keeping things general would be beneficial.  For one thing, the tracking branch is actually created with `checkout -b`, so in this bullet point, `fetch` is just working with a remote intended for tracking (IIUC).  Also, for someone naively following a step-by-step (my case), this general description would have been enough, and maybe more understandable than the introduction of a new concept.

I'll go ahead and post a new commit.  FYI, I merged upstream/master and I think I did it correctly but please let me know if I should fix anything on my end.